### PR TITLE
fix(scanv4): Change charts so that Scanner V4 becomes opt-in

### PR DESCRIPTION
**Pull request by mclasmeier**
## Description

As aligned, this PR changes the Scanner V4 integration so that it becomes opt-in.

Can be reviewed commit-by-commit.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [x] ~~Evaluated and added CHANGELOG entry if required~~ (will be done separately)
- [x] ~~Determined and documented upgrade steps~~ (none)
- [x] ~~Documented user facing changes~~ (docs are done outside of this ticket)

## Testing Performed

CI (unit tests plus installation e2e tests).

### CODE REVIEW

**Comment by mclasmeier** [comment]
@vladbologa @johannes94 FYI, I believe the test failures are not related to the change.

File: pkg/renderer/templates/public_values.yaml.tpl [hunk 1]
```diff
@@ -166,6 +166,7 @@ scanner:
   {{- end }}
 
 scannerV4:
+  disable: false
```
Comment (vladbologa) [inline pkg/renderer/templates/public_values.yaml.tpl:169]: Won't this enable scanner V4 if you run `helm [...] -f values-public.yaml`?

**Review by johannes94 — COMMENTED** [review]
Looks good, would approve as soon as the public_values.yaml.tpl mentioned by Vlad is fixed.

File: pkg/renderer/templates/public_values.yaml.tpl (hunk 1 above)
Comment (mclasmeier) [reply pkg/renderer/templates/public_values.yaml.tpl]: Thank you, good catch. Just pushed a commit to fix this, let's see what CI thinks about it.

**Review by johannes94 — APPROVED** [review]

**Pull request merged**